src/amo/components/Home/styles.scss

```diff
@@ -58,6 +58,23 @@
   }
 }
 
+.Home-SubjectShelf-link-text {
+  text-align: left;
+  width: calc(
+    100% - 70px
+  ); /* 70px accounts for approx width of Icon plus some cushion */
+
+  @include respond-to(medium) {
+    width: calc(
+      100% - 100px
+    ); /* 100px accounts for approx width of Icon plus some cushion */
+  }
+
+  @include respond-to(large) {
+    width: auto;
+  }
+}
```

Comment:
hi @hritvi : ) I am not sure you need any of this anymore if you're not using [flex-grow](https://github.com/mozilla/addons-frontend/pull/5974#discussion_r215327406) here. So I think you can remove all this, as it doesn't seem to do anything now. This means you can remove the class (Home-SubjectShelf-link-text) on the span above. 

Please let me know if you see something different..